rtgui/cropwindow.cc

```diff
@@ -272,210 +269,209 @@ void CropWindow::buttonPress (int button, int type, int bstate, int x, int y)
 {
 
     bool needRedraw = true;  // most common case ; not redrawing are exceptions
+    EditSubscriber *editSubscriber = iarea->getCurrSubscriber();
 
     iarea->grabFocus (this);
 
-    if (button == 1 && type == GDK_2BUTTON_PRESS && onArea (CropImage, x, y) && (state == SNormal || state == SCropImgMove)) {
-        if (fitZoomEnabled) {
-            if (fitZoom) {
-                state = SNormal;
-                zoomVersion = exposeVersion;
-                screenCoordToImage (x, y, action_x, action_y);
-                changeZoom (ZOOM11INDEX, true, action_x, action_y);
-                fitZoom = false;
+    if (button == 1) {
```

Comment:
Well, it probably isn't worth it to try to make this data-driven: At the end of the day, this wants to use widgets or a scene graph or something that allows to dispatch the events and handle them locally within a separate object. This is just how GUI code looked like before OOP. Maybe we can someday find the time to integrate this with the rest of the scene graph infrastructure.